.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,120 @@
+  # without comprimizing the security
+  upload-event-file:
+    name: Upload Event File
+    runs-on: ubuntu-latest
+    steps:
+      - name: Upload event file
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-event-file
+          path: ${{ github.event_path }}
+          retention-days: 1
+
+  # https://stackoverflow.com/a/77549656
+  variables:
+    name: Variable Accessibility Workaround for Jobs
+    runs-on: ubuntu-latest
+    outputs:
+      EXPERIMENTAL: ${{ env.EXPERIMENTAL == 'true' || (github.event == 'push' && github.ref == 'refs/heads/master' || github.ref == 'refs/heads/development') }}
+      BUILD_CONFIGURATIONS: ${{env.BUILD_CONFIGURATIONS}}
+      DOTNET_TARGET_FRAMEWORKS: ${{env.DOTNET_TARGET_FRAMEWORKS}}
+      DOTNET_TARGET_FRAMEWORKS_MACOS_ARM64: ${{env.DOTNET_TARGET_FRAMEWORKS_MACOS_ARM64}}
+      FRAMEWORK_TARGET_FRAMEWORKS: ${{env.FRAMEWORK_TARGET_FRAMEWORKS}}
+    steps:
+      - name: Compute outputs
+        run: |
+          echo "EXPERIMENTAL=${{env.EXPERIMENTAL}}" >> $GITHUB_OUTPUT
+          echo "BUILD_CONFIGURATIONS=${{env.BUILD_CONFIGURATIONS}}" >> $GITHUB_OUTPUT
+          echo "DOTNET_TARGET_FRAMEWORKS=${{env.DOTNET_TARGET_FRAMEWORKS}}" >> $GITHUB_OUTPUT
+          echo "DOTNET_TARGET_FRAMEWORKS_MACOS_ARM64=${{env.DOTNET_TARGET_FRAMEWORKS_MACOS_ARM64}}" >> $GITHUB_OUTPUT
+          echo "FRAMEWORK_TARGET_FRAMEWORKS=${{env.FRAMEWORK_TARGET_FRAMEWORKS}}" >> $GITHUB_OUTPUT
+
+  build:
+    name: Perform Build
+    needs: [variables]
+    strategy:
+      matrix:
+        runner: [ 
+          { image: 'windows-latest', arch: 'x64' },
+          { image: 'windows-latest', arch: 'x86' },
+        ]
+        build_configuration: ${{fromJson(needs.variables.outputs.BUILD_CONFIGURATIONS)}}
+    uses: ./.github/workflows/reusable-build.yml
+    with:
+      image: ${{matrix.runner.image}}
+      architecture: ${{matrix.runner.arch}}
+      build_configuration: ${{matrix.build_configuration}}
+
+  test:
+    name: Execute Tests
+    needs: [variables, build]
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: [ 
+          { image: 'windows-latest', arch: 'x64' },
+          { image: 'windows-latest', arch: 'x86' },
+          { image: 'ubuntu-latest', arch: 'x64' },
+          { image: 'macos-13', arch: 'x64' },
+          { image: 'macos-14', arch: 'arm64' },
+          { image: 'macos-14', arch: 'x64' },
+        ]
+        build_configuration: ${{fromJson(needs.variables.outputs.BUILD_CONFIGURATIONS)}}
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      image: ${{matrix.runner.image}}
+      architecture: ${{matrix.runner.arch}}
+      dotnet_target_frameworks: ${{needs.variables.outputs.DOTNET_TARGET_FRAMEWORKS}}
+      build_configuration: ${{matrix.build_configuration}}
+      #ignore non-windows tests since support is yet lacking
+      is_experimental: ${{matrix.runner.image != 'windows-latest'}}
```

Comment:
713853a30838d753f02fe8284ef27e5201a68015 should make it possible to change this to macos only.

Reply:
Done, Ubuntu not passes tests